source/kernel/src/daemons/shells.rs

```diff
@@ -19,43 +27,150 @@ use profont::PROFONT_12_POINT;
+        Self {
+            port: WellKnown::ForthShell0.into(),
+            capacity: 256,
+            forth_settings: Default::default(),
+            _priv: (),
+        }
+    }
+}
+
+/// Spawns a forth shell on the given port
 #[tracing::instrument(skip(k))]
-pub async fn graphical_shell_mono(
-    k: &'static Kernel,
-    disp_width_px: u32,
-    disp_height_px: u32,
-    port: u16,
-    capacity: usize,
-) {
+pub async fn sermux_shell(k: &'static Kernel, settings: SermuxShellSettings) {
+    let SermuxShellSettings {
+        port,
+        capacity,
+        forth_settings,
+        _priv,
+    } = settings;
+    let port = PortHandle::open(k, port, capacity).await.unwrap();
+    let (task, tid_io) = Forth::new(k, forth_settings)
+        .await
+        .expect("Forth spawning must succeed");
+    k.spawn(task.run()).await;
+    k.spawn(async move {
+        loop {
+            futures::select_biased! {
+                rgr = port.consumer().read_grant().fuse() => {
+                    let needed = rgr.len();
+                    let mut tid_io_wgr = tid_io.producer().send_grant_exact(needed).await;
+                    tid_io_wgr.copy_from_slice(&rgr);
+                    tid_io_wgr.commit(needed);
+                    rgr.release(needed);
+                },
+                output = tid_io.consumer().read_grant().fuse() => {
+                    let needed = output.len();
+                    port.send(&output).await;
+                    output.release(needed);
+                }
+            }
+        }
+    })
+    .await;
+}
+
+/// Settings for the [graphical_shell_mono] daemon
+///
+/// This does NOT implement [Default]. Instead use [GraphicalShellSettings::with_display_size].
+#[derive(Debug)]
+pub struct GraphicalShellSettings {
+    /// Sermux port to use as a PsuedoKeyboard.
+    ///
+    /// Defaults to [WellKnown::PsuedoKeyboard]
+    pub port: u16,
+    /// Number of bytes used for the sermux buffer
+    ///
+    /// Defaults to 256
+    pub capacity: usize,
+    /// Forth parameters for the shell
+    ///
+    /// Uses the default value of [Params]
+    pub forth_settings: Params,
+    /// Display width in pixels
+    pub disp_width_px: u32,
+    /// Display height in pixels
+    pub disp_height_px: u32,
+    /// Font used for the shell
+    ///
+    /// Defaults to [PROFONT_12_POINT]
+    pub font: MonoFont<'static>,
+    /// Hidden for forwards compat
+    _priv: (),
+}
+
+impl GraphicalShellSettings {
+    pub fn with_display_size(width_px: u32, height_px: u32) -> Self {
+        Self {
+            port: WellKnown::PsuedoKeyboard.into(),
+            capacity: 256,
+            forth_settings: Default::default(),
+            disp_width_px: width_px,
+            disp_height_px: height_px,
+            font: PROFONT_12_POINT,
+            _priv: (),
+        }
+    }
+}
+
+/// Spawns a graphical shell using the [EmbDisplayService][crate::services::emb_display::EmbDisplayService] service
```

Comment:
i think the rustdoc link syntax is wrong here, and this should be:
```suggestion
/// Spawns a graphical shell using the [EmbDisplayService](crate::services::emb_display::EmbDisplayService]) service
```